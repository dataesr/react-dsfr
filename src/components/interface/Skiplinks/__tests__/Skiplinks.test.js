import { render, screen, within } from '@testing-library/react';
import { SkiplinkItem, Skiplinks } from '..';

describe('<SkipLinks />', () => {
  it('should renders SkipLinks properly', () => {
    render(
      <Skiplinks data-testid="skiplinks">
        <SkiplinkItem href="#" data-testid="skiplinkitem1">Accéder au contenu</SkiplinkItem>
        <SkiplinkItem href="#" data-test-id="skiplinkitem2">Accéder au menu</SkiplinkItem>
        <SkiplinkItem href="#" data-test--id="skiplinkitem3">Accéder à la recherche</SkiplinkItem>
        <SkiplinkItem href="#" data-test="skiplinkitem4">Accéder au footer</SkiplinkItem>
      </Skiplinks>,
    );
    const skiplinks = screen.getByTestId('skiplinks');
    expect(skiplinks).toBeInTheDocument();
    expect(skiplinks).toMatchSnapshot();
    const links = screen.getByRole('list');
    const { getAllByRole } = within(links);
    const items = getAllByRole('listitem');
    expect(items.length).toBe(4);
  });
});
