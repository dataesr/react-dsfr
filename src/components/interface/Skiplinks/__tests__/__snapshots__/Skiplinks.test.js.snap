// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipLinks /> should renders Tag properly 1`] = `
<div
  class="fr-skiplinks"
  data-testid="skiplinks"
>
  <div
    class="fr-container"
  >
    <ul
      class="fr-skiplinks__list"
    >
      <li
        class=""
      >
        <a
          class="fr-link"
          href="#"
          target="_self"
        >
          Accéder au contenu
        </a>
      </li>
      <li
        class=""
      >
        <a
          class="fr-link"
          href="#"
          target="_self"
        >
          Accéder au menu
        </a>
      </li>
      <li
        class=""
      >
        <a
          class="fr-link"
          href="#"
          target="_self"
        >
          Accéder à la recherche
        </a>
      </li>
      <li
        class=""
      >
        <a
          class="fr-link"
          href="#"
          target="_self"
        >
          Accéder au footer
        </a>
      </li>
    </ul>
  </div>
</div>
`;
